streamalert/artifact_extractor/artifact_extractor.py

```diff
@@ -39,14 +37,16 @@ def __init__(self, source_type, normalized_type, value, **kwargs):
                 could have a value of “50.50.50.50”
 
         Kwargs (optional):
-            function (str): Describes how this field is used in the record, or what it means.
+            function (str): Describes how this field is used in the record, or what it means. It
+                will be "None" if no "function" information when searching artifacts in Athena.
             record_id (str): Currently it is reserved for future support. It will come from the
                 record processed by classifier. This field is very useful for cross reference back
-                to the original record in the future.
+                to the original record in the future. It will be "None" if no "record_id"
+                information when searching artifacts in Athena.
         """
         # Enforce all fields are strings in a Artifact to prevent type corruption in Parquet format
-        self._function = str(kwargs.get('function', 'not_specified'))
-        self._record_id = str(kwargs.get('record_id', self.RESERVED))
+        self._function = str(kwargs.get('function'))
```

Comment:
if these values ('function' and 'record_id') are no longer optional, please remove the usage of `kwargs` and add these are required arguments to `__init__`